### PR TITLE
Use generateDS to build xml document in Python

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
generateDS is run with the following command:

```console
python .\generateDS\generateDS.py -o w3d_translator/generateDS/classes.py -s w3d_translator/generateDS/subclasses.py --super=generateDS.classes --use-getter-setter=none --member-specs=dict --cleanup-name-list="[('[-:.]', '_'), ('Root$', 's'), ('Pos$', '')]" .\schema\caveschema.xsd
```

- Cleans `caveschema.xsd` file
- Adds `generateDS` folder in module (output of `generateDS` script)
- Updates README with information about `generateDS`
- closes issue #6
- closes issue #7
- closes issue #8 
- closes issue #9 
- closes issue #10 
- closes issue #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,10 +17,41 @@ The translator uses Unity version 2021.3.0f1. Please ensure the `Unity.exe` file
...
+
+## generateDS
+
+This project uses [generateDS](https://www.davekuhlman.org/generateDS.html) to build each xml file in Python. This intermediary step is done to ease the translation into the Unity yaml specification. The relevant files have already been generated and changed, and should not need to be touched.
+
+Should you need to regenerate the code, generateDS should be [installed](https://www.davekuhlman.org/generateDS.html#how-to-build-and-install-it) on your local machine. It is run with the following arguments:
+
+```console
+python generateDS.py -o w3d_translator/generateDS/[classes.py] -s w3d_translator/generateDS/[subclasses.py] --super=[generateDS.classes] --use-getter-setter=none --member-specs=dict --cleanup-name-list="[('[-:.]', '_'), ('Root$', 's'), ('Pos$', '')]" .\schema\caveschema.xsd
+```
```
[the remark this answers is not in the document]
Comment (broarr) [reply README.md:44]: Below you're installing `generateDS` in poetry, does that mean the modules are callable from the poetry environment?

File: w3d_translator/__init__.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 __version__ = "0.1.0"
 
-from .read_xml import parse_string
+from .translate import parse_string
```
Comment (broarr) [inline w3d_translator/__init__.py:3]: I thought relative imports have been deprecated?

**Review by broarr — COMMENTED** [review]
I'm a little confused on this PR. Should the `generateDS` files even be in this one? Can you do everything with `generateDS` from poetry given that you're installing `generateDS` into your poetry env?

File: .gitignore [hunk 3]
```diff
@@ -63,4 +63,4 @@ crashlytics-build.properties
 __pycache__/
 *.pyc
 out/
-/generateDS/
\ No newline at end of file
+/generateDS/
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply .gitignore:66]: Yes, I did it that way before I used poetry. I'll definitely try running the command with poetry and seeing if that works. If it does I'll delete the source code and then this ignore

File: schema/caveschema.xsd [hunk 4]
```diff
@@ -11,10 +11,9 @@
         <xs:element name="ObjectRoot" minOccurs="0" maxOccurs="1">
           <xs:complexType>
             <xs:sequence>
-              <xs:element ref="Object" maxOccurs="unbounded" />
+              <xs:element ref="Object" minOccurs="0" maxOccurs="unbounded" />
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply schema/caveschema.xsd:14]: I mean several of the test projects were failing (including the main example John gave me) before I made the change. I suppose it's a matter of if I should be fixing those projects to conform or vice versa? I kept a copy of the original so we still have it if we really need

[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply schema/caveschema.xsd:14]: I definitely need to keep the change to the `xsd:key` and `xsd:keyref` components though - those weren't validating correctly in the original schema

File: w3d_translator/generateDS/custom_methods.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+# -*- mode: pymode; coding: latin1; -*-
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply w3d_translator/generateDS/custom_methods.py:2]: Yes, although if it's easy enough to run it with poetry then perhaps I won't commit them

File: w3d_translator/__init__.py (hunk 2 above)
Comment (hollandjg) [reply w3d_translator/__init__.py:3]: I think they might still be okay – at least the documentation says they're fine (as we're inside a package) – or have I missed something? https://docs.python.org/3/reference/import.html#package-relative-imports

Comment (RobertGemmaJr) [reply w3d_translator/__init__.py:3]: It still works fine as `from translate import parse_string` so I'll make the change

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply README.md:44]: Getting this error when I run `poetry run generateDS`

```console
Traceback (most recent call last):
  File "C:\Users\Rob\.pyenv\pyenv-win\versions\3.10.0\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Rob\.pyenv\pyenv-win\versions\3.10.0\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\Rob\ROOT\CCV\W3D Translator\.venv\Scripts\generateDS.exe\__main__.py", line 4, in <module>
ImportError: cannot import name 'main' from 'generateDS' (unknown location)
```

I also tried `poetry run generateDS.py` and `poetry run python generateDS(.py)` with no luck. The documentation talks about downloading the source code directly so maybe it really is the only way to go?

[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply README.md:44]: https://www.davekuhlman.org/generateDS.html#where-to-find-it

[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply README.md:44]: Okay so it looks like I can run it via `poetry` but I have to give the explicit path:

```console
poetry run python .\.venv\Scripts\generateDS.py -o w3d_translator/generateDS/classes.py -s w3d_translator/generateDS/subclasses.py --super=[generateDS.classes] --use-getter-setter=none --member-specs=dict --cleanup-name-list="[('[-:.]', '_'), ('Root$', 's'), ('Pos$', '')]" .\schema\caveschema.xsd
```

Does that seem like an appropriate way to do things? I tried cloning again and when I ran `poetry install` it built the `.venv` folder so I think it's okay?

File: .gitignore [hunk 6]
```diff
@@ -63,4 +63,3 @@ crashlytics-build.properties
 __pycache__/
 *.pyc
 out/
-/generateDS/
```
Comment (broarr) [inline .gitignore:66]: I think we still want this directory ignored right? We shouldn't be committing third-party code. Those should be installed via other methods

**Review by broarr — APPROVED** [review]
Do we need the `generateDS` code?

File: .gitignore (hunk 6 above)
Comment (broarr) [reply .gitignore:66]: Ahh, I think I was confused. Is this `generateDS` directory for the generated files?

**Pull request merged**